tests/manage/pv_services/test_create_pvc_random_storage_class.py

```diff
@@ -8,128 +8,68 @@
-    for sc_count in range(count):
-        log.info("Creating CephBlockPool")
-        POOL_OBJS[sc_count] = helpers.create_ceph_block_pool()
-        SC_RBD_OBJS[sc_count] = helpers.create_storage_class(
-            constants.CEPHBLOCKPOOL,
-            interface_name=POOL_OBJS[sc_count].name,
-            secret_name=RBD_SECRET_OBJ.name
-        )
-
-    return True
-
-
-def create_pvc(storageclass_list, count=1):
-    """
-    Function for creating pvc and multiple pvc
-
-    Args:
-        storageclass_list (list): This will contain storageclass list
-        count (int): count specify no of pvc want's to create
-    """
-    global PVC_OBJS
-    PVC_OBJS = [0] * count
-    for i in range(count):
-        sc_name = random.choice(storageclass_list)
-        PVC_OBJS[i] = helpers.create_pvc(sc_name)
-    for pvc_obj in PVC_OBJS:
-        helpers.wait_for_resource_state(pvc_obj, constants.STATUS_BOUND)
-        pvc_obj.reload()
-    return True
-
-
-def create_storageclass_cephfs():
-    """
-    Function for creating CephFs storageclass
-    """
-    global SC_CEPHFS_OBJ
-    SC_CEPHFS_OBJ = helpers.create_storage_class(
-        constants.CEPHFILESYSTEM,
-        helpers.get_cephfs_data_pool_name(),
-        CEPHFS_SECRET_OBJ.name
-    )
+from tests.fixtures import (
+    create_rbd_secret, create_cephfs_secret
+)
 
-    return True
+log = logging.getLogger(__name__)
 
 
 @tier2
 @pytest.mark.usefixtures(
-    ocs288_fixture.__name__,
+    create_rbd_secret.__name__,
+    create_cephfs_secret.__name__,
 )
 @pytest.mark.polarion_id("OCS-288")
 class TestCreatePVCRandomStorageClass(ManageTest):
     """
     Creating PVC with random SC
     """
 
-    def test_create_pvc_with_random_sc(self):
-        storageclass_list = helpers.get_all_storageclass_names()
-        if len(storageclass_list):
-            assert create_pvc(storageclass_list, count=20)
-        else:
-            log.error("No Storageclass Found")
+    def test_create_pvc_with_random_sc(self, teardown_factory):
+        sc_list = []
+        for i in range(5):
+            log.info(f"Creating cephblockpool")
+            cbp_obj = helpers.create_ceph_block_pool()
+            log.info(
+                f"{cbp_obj.name} created successfully"
+            )
+            log.info(
+                f"Creating a RBD storage class using {cbp_obj.name}"
+            )
+            rbd_sc_obj = helpers.create_storage_class(
+                interface_type=constants.CEPHBLOCKPOOL,
+                interface_name=cbp_obj.name,
+                secret_name=self.rbd_secret_obj.name
+            )
+            log.info(
+                f"StorageClass: {rbd_sc_obj.name} "
+                f"created successfully using {cbp_obj.name}"
+            )
+            sc_list.append(rbd_sc_obj)
+            teardown_factory(cbp_obj)
+            teardown_factory(rbd_sc_obj)
+        log.info("Creating CephFs Storageclass")
+        cephfs_sc_obj = helpers.create_storage_class(
```

Comment:
only 1 FS storage class is being created, while above (line 50), 5 RBD pools are being appended. Intentional?

Reply:
yes because for FS sc every time interface_name will remain same i.e fs_poo_data_name so i am creating 1 cephfs sc